crates/bifrost/src/appender.rs

```diff
@@ -169,19 +183,67 @@ impl Appender {
-                    total_dur
-                );
+                if tone_escalated {
+                    info!(
+                        open_segment = %loglet.segment_index(),
+                        "Reconfiguration complete, found an open segment after {:?}",
+                        total_dur
+                    );
+                } else {
+                    info!(
+                        open_segment = %loglet.segment_index(),
+                        "Reconfiguration complete, found an open segment after {:?}",
+                        total_dur
+                    );
+                }
                 return Ok(loglet);
             } else {
                 let log_version = Metadata::with_current(|m| m.logs_version());
-                debug!(
-                    log_version = %log_version,
-                    "Still waiting for sealing to complete. Elapsed={:?}",
-                    start.elapsed(),
-                );
+                if start.elapsed() > recovery_patience_dur {
+                    tone_escalated = true;
+
+                    if error_recovery_strategy >= ErrorRecoveryStrategy::ExtendChainAllowed {
+                        // take the matter in our own hands
+                        let admin = BifrostAdmin::new(bifrost_inner);
+                        info!(
+                            %sealed_segment,
+                            "[Auto Recovery] Attempting to extend the chain to allow new appends to proceed on a new configuration. We waited for {:?} before triggering automatic recovery",
+                            start.elapsed(),
+                        );
+                        if let Err(err) = admin
+                            .seal_and_auto_extend_chain(log_id, Some(sealed_segment))
+                            .log_slow_after(
+                                recovery_patience_dur / 2,
+                                tracing::Level::INFO,
+                                "Extending the chain with new configuration",
+                            )
+                            .with_overdue(recovery_patience_dur, tracing::Level::WARN)
+                            .await
+                        {
+                            // we couldn't reconfigure. Let the loop handle retries as normal
+                            info!(
+                                %err,
+                                %log_version,
+                                "Could not reconfigure the log, perhaps someone else beat us to it? We'll check",
+                            );
+                        }
+                    } else {
+                        // Holding pattern
+                        if tone_escalated {
+                            info!(
+                                %log_version,
+                                "In holding pattern, still waiting for log reconfiguration to complete. Elapsed={:?}",
+                                start.elapsed(),
+                            );
```

Comment:
Wouldn't we always be in the outer if branch if `tone_escalated` is true?

Reply:
That wasn't correct. The new version looks different. Can you please re-check?